src/server/handlers/launchpad.js

```diff
@@ -411,34 +412,60 @@ export const internalFindSnap = async (repositoryUrl) => {
   throw new PreparedError(404, RESPONSE_SNAP_NOT_FOUND);
 };
 
-const internalFindSnapsByPrefix = async (urlPrefix) => {
+const internalFindSnaps = async (owner, token) => {
+  const orgs = await internalListOrganizations(owner, token);
+  const owners = [owner].concat(orgs.map((org) => org.login));
+  const urlPrefixes = owners.map(getRepoUrlPrefix);
   const username = conf.get('LP_API_USERNAME');
-  const cacheId = getUrlPrefixCacheId(urlPrefix);
+  const cacheIds = urlPrefixes.map(getUrlPrefixCacheId);
+  const memcached = getMemcached();
   const lpClient = getLaunchpad();
+  let snaps;
+  let remainingPrefixes;
 
   try {
-    const result = await getMemcached().get(cacheId);
-    if (result !== undefined) {
-      return result.map((entry) => {
+    snaps = await Promise.all(
+      cacheIds.map((cacheId) => memcached.get(cacheId))
+    );
+    remainingPrefixes = zip(urlPrefixes, snaps)
+      .filter(([, snap]) => snap === undefined)
+      .map(([urlPrefix, ]) => urlPrefix);
```

Comment:
Consider a new line here, so I can catch my breath :)

(I think the old style guide said new line after assignments)